packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -619,8 +619,8 @@ abstract class Page<T> extends RouteSettings {
 /// An interface for observing the behavior of a [Navigator].
 class NavigatorObserver {
   /// The navigator that the observer is observing, if any.
-  NavigatorState? get navigator => _navigator;
-  NavigatorState? _navigator;
+  NavigatorState? get navigator => _navigators[this];
+  static final _navigators = Expando<NavigatorState>();
```

Comment:
Could you leave a comment on this Expando property describing why we are doing this. I am sure the next guy looking at this will be somewhat confused as to why we do this here.

Reply:
Done.